### PR TITLE
HB-5754: SwiftUI demo app

### PR DESCRIPTION
**Pull request by bwised**
https://chartboost.atlassian.net/browse/HB-5754

### CODE REVIEW

File: swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+
+private struct AdAnchor: UIViewControllerRepresentable {
+    typealias UIViewControllerType = UIViewController
+
+    let didMake: (UIViewController) -> ()
+
+    func makeUIViewController(context: Context) -> UIViewController {
+        let viewController = UIViewController()
```
Comment (daniel-barros) [inline swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift:33]: I'm not sure this is the best way to provide the view controller needed to show ads.
If I'm not wrong we are never presenting this view controller on screen, which means it is a detached vc. Presenting vcs on top of detached vcs may lead to errors. I remember getting some console warnings about this in the past, but correct me if my knowledge is outdated.

Maybe we could use the top-most view controller instead? We could have the InterstitialAdController/RewardedAdController/BannerAdController class obtain it so SwiftUI views don't need to care about vcs or anchors.
This is how it was solved in the SwiftUI demo app the ads sdk team built. This is a snippet I stole from that codebase:
```
extension UIApplication {
    class func topViewController(base: [UIWindow] = UIApplication.shared.windows) -> UIViewController? {
        let keyWindow = base.filter {$0.isKeyWindow}.first
        if var topViewController = keyWindow?.rootViewController {
            while let presentedViewController = topViewController.presentedViewController {
                topViewController = presentedViewController
            }
            return topViewController
        }
        return nil
    }
}
```

Comment (bwised) [reply swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift:33]: Cool, I'll give it a try.

Comment (bwised) [reply swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift:33]: @daniel-barros Question.  If you feel that using that extension is the way to go, then why doesn't the Chartboost Mediation SDK just do something like that internally and not require view controllers to be passed into the APIs at all?

Comment (bwised) [reply swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift:33]: This is more modern:

```
    class var topViewController: UIViewController? {
        guard
            let windowScene = UIApplication.shared.connectedScenes.compactMap({ $0 as? UIWindowScene }).first,
            let keyWindow = windowScene.keyWindow,
            var rootViewController = keyWindow.rootViewController
        else {
            return nil
        }
        while let presentedViewController = rootViewController.presentedViewController {
            rootViewController = presentedViewController
        }
        return rootViewController
    }
```

Comment (daniel-barros) [reply swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift]: So this solution is not perfect either for UIKit apps. Some sophisticated users may have multiple windows (if I remember right you can windows inside windows although I don't know why you'd do that), also there can be multiple scenes... So it's safer to let pubs define the VC themselves, this is what most SDKs do.

Feel free to do more research on this issue. My suggestion is likely not the definitive solution. Or we can create a ticket to do the research.

Comment (bwised) [reply swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift]: Well, we know of 2 ways for sure (my original way and this other way), both of which would work fine for this demo.  Do we really need to know more?

Comment (daniel-barros) [reply swiftui/ChartboostMediationDemo/Ads/AdAnchor.swift]: Yeah I think this is good enough for now given that SwiftUI pubs are not a sizable part of our user base.

**Review by daniel-barros — APPROVED** [review]

**Pull request merged**